src/Data/Machine/Mealy.purs

```diff
@@ -48,11 +49,24 @@ import Effect.Class (class MonadEffect, liftEffect)
 
 newtype MealyT f s a = MealyT (s -> f (Step f s a))
 
+-- |let's you transform a mealy machine running in the context of `f` into one running in `g`
+-- if you can give a natural transformation from `f` to `g`
+-- 
+-- this can be useful if you use monad-transformers for example
+hoistMealy :: forall f g s a . Functor g => (f ~> g) -> MealyT f s a -> MealyT g s a
+hoistMealy f2g (MealyT goF) = MealyT goG
+  where goG s = hoistStep f2g <$> f2g (goF s)
+
 runMealyT :: forall f s a. MealyT f s a -> s -> f (Step f s a)
 runMealyT (MealyT f) = f
 
 data Step f s a = Emit a (MealyT f s a) | Halt
 
+hoistStep :: forall f g s a. Functor g => (f ~> g) -> Step f s a -> Step g s a
```

Comment:
Not super important, but these hoist functions could use `~>` on the `Step` / `MealyT` too, dropping the `a`.